File: src/python/finished/binary_gap.py
# https://leetcode.com/problems/binary-gap/


class Solution:
    def binaryGap(self, n: int) -> int:
        prev_one = 64
        cur_idx = 0

        best = 0
        cur = n
        while cur:
            bit = cur & 1
            cur >>= 1
            if bit == 1:
                best = max(best, cur_idx - prev_one)
                prev_one = cur_idx
            cur_idx += 1

        return best

File: src/rust/finished/binary_gap.rs
impl Solution {
    pub fn binary_gap(n: i32) -> i32 {
        let mut prev_one_idx = 1 << 16;
        let mut cur_idx = 0;

        let mut best = 0;
        let mut cur = n;
        while cur != 0 {
            let bit = cur & 1;
            cur >>= 1;

            if bit == 1 {
                best = best.max(cur_idx - prev_one_idx);
                prev_one_idx = cur_idx;
            }

            cur_idx += 1;
        }

        best
    }
}
